compiler/rustc_typeck/src/check/method/mod.rs

```diff
@@ -342,7 +344,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         // Now we want to know if this can be matched
         if !self.predicate_may_hold(&obligation) {
             debug!("--> Cannot match obligation");
-            return None; // Cannot be matched, no such method resolution is possible.
+            let mut fulfill = <dyn traits::TraitEngine<'_>>::new(self.tcx);
+            fulfill.register_predicate_obligation(self, obligation);
+            // Cannot be matched, no such method resolution is possible, but we collect the
+            // unfulfilled obligations for better error reporting.
+            return Err(fulfill.select_where_possible(&self.infcx).unwrap_err());
```

Comment:
`lookup_method_in_trait` can be tried on many traits, so this code is run on a good path and may affect performance even if no errors are reported.

Reply:
Is it possible to return `Err(obligation)` instead, and perform these actions only if the error is actually emitted?